src/main/resources/migration/changelog-v5.6.0.xml

```diff
@@ -107,4 +107,202 @@
+            <column name="CLASSICAL_SECURITY_LEVEL" type="INT"/>
+            <column name="NIST_QUANTUM_SECURITY_LEVEL" type="INT"/>
+        </createTable>
+
+        <createTable tableName="CERTIFICATE_PROPERTIES">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="CERTIFICATE_PROPERTIES_PK"/>
+            </column>
+            <column name="SUBJECT_NAME" type="VARCHAR(255)"/>
+            <column name="ISSUER_NAME" type="VARCHAR(255)"/>
+            <column name="NOT_VALID_BEFORE" type="TIMESTAMP WITH TIME ZONE"/>
+            <column name="NOT_VALID_AFTER" type="TIMESTAMP WITH TIME ZONE"/>
+            <column name="SIGNATURE_ALGORITHM_REF" type="VARCHAR(64)"/>
+            <column name="SUBJECT_PUBLIC_KEY_REF" type="VARCHAR(64)"/>
+            <column name="CERTIFICATE_FORMAT" type="VARCHAR(32)"/>
+            <column name="CERTIFICATE_EXTENSION" type="VARCHAR(32)"/>
+        </createTable>
+
+        <createTable tableName="RELATED_CRYPTO_MATERIAL_PROPERTIES">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="RELATED_CRYPTO_MATERIAL_PROPERTIES_PK"/>
+            </column>
+            <column name="TYPE" type="VARCHAR(32)"/>
+            <column name="IDENTIFIER" type="VARCHAR(64)"/>
+            <column name="STATE" type="VARCHAR(16)"/>
+            <column name="ALGORITHM_REF" type="VARCHAR(64)"/>
+            <column name="CREATION_DATE" type="TIMESTAMP WITH TIME ZONE"/>
+            <column name="ACTIVATION_DATE" type="TIMESTAMP WITH TIME ZONE"/>
+            <column name="UPDATE_DATE" type="TIMESTAMP WITH TIME ZONE"/>
+            <column name="EXPIRATION_DATE" type="TIMESTAMP WITH TIME ZONE"/>
+            <column name="VALUE" type="VARCHAR(32)"/>
+            <column name="SIZE" type="INT"/>
+            <column name="FORMAT" type="VARCHAR(8)"/>
+            <column name="SECURED_BY_MECHANISM" type="VARCHAR(16)"/>
+            <column name="SECURED_BY_ALGORITHM_REF" type="VARCHAR(64)"/>
+        </createTable>
+
+        <createTable tableName="PROTOCOL_PROPERTIES">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="PROTOCOL_PROPERTIES_PK"/>
+            </column>
+            <column name="TYPE" type="VARCHAR(16)"/>
+            <column name="VERSION" type="VARCHAR(16)"/>
+        </createTable>
+
+        <createTable tableName="CRYPTO_FUNCTIONS">
+            <column autoIncrement="true" name="ID" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="CRYPTO_FUNCTIONS_PK"/>
+            </column>
+            <column name="ALGORITHM_PROPERTY_ID" type="BIGINT"/>
+            <column name="CRYPTO_FUNCTION" type="VARCHAR(32)"/>
+        </createTable>
```

Comment:
Yes, absolutely. I avoided array column types since they are not supported by all db systems. But if we stick to postgres using array types would greatly simplify the table structure. There are a couple of similar structures in CBOM 1.6 where we could migrate to arrays.